spyder/plugins/ipythonconsole/widgets/tests/test_kernelconnect.py

```diff
@@ -0,0 +1,201 @@
+            # fill out port
+            dlg.pn.clear()
+            qtbot.keyClicks(dlg.pn, str(pn))
+            # fill out username
+            dlg.un.clear()
+            qtbot.keyClicks(dlg.un, un)
+
+            if use_keyfile:
+                # select ssh keyfile radio
+                dlg.kf_radio.setChecked(True)
+                assert dlg.kf.isEnabled()
+
+                # fill out ssh keyfile
+                dlg.kf.clear()
+                qtbot.keyClicks(dlg.kf, kf)
+
+                # fill out passphrase
+                dlg.kfp.clear()
+                qtbot.keyClicks(dlg.kfp, kfp)
+            else:
+                # select password radio
+                dlg.pw_radio.setChecked(True)
+                assert dlg.pw.isEnabled()
+
+                # fill out password
+                dlg.pw.clear()
+                qtbot.keyClicks(dlg.pw, pw)
+
+            # check save connection settings
+            dlg.save_layout.setChecked(save_settings)
+
+            # Press ok and save connection settings
+            qtbot.mouseClick(dlg.accept_btns.button(QDialogButtonBox.Ok),
+                             Qt.LeftButton)
+
+            return dlg
+
+    def teardown():
+        """Clear existing-kernel config and keyring passwords"""
+        CONF.remove_section("existing-kernel")
+
+        try:
+            import keyring
+            keyring.set_password("existing_kernel",
+                                 "ssh_key_passphrase", "")
+            keyring.set_password("existing_kernel",
+                                 "ssh_password", "")
+        except Exception:
+            pass
+
+    request.addfinalizer(teardown)
+    return DialogFactory()
+
+
+cf_path = "test_cf_path"
+un = "test_username"
+hn = "test_hostname"
+pn = 123
+kf = "test_kf"
+kfp = "test_kfp"
+pw = "test_pw"
+
+
+# =============================================================================
+# Tests
+# =============================================================================
+def test_kernel_connection_dialog_remember_input_ssh_passphrase(
+        qtbot, connection_dialog_factory):
+    """Test that the dialog remember user's kernel connection
+       settings and ssh key passphrase when the user checks the
+       save checkbox."""
+
+    connection_dialog_factory.submit_filled_dialog(use_keyfile=True,
+                                                 save_settings=True)
+
+    # create new dialog and check fields
+    new_dlg = connection_dialog_factory.get_default_dialog()
+    assert new_dlg.cf.text() == cf_path
+    assert new_dlg.rm_group.isChecked()
+    assert new_dlg.hn.text() == hn
+    assert new_dlg.un.text() == un
+    assert new_dlg.pn.text() == str(pn)
+    assert new_dlg.kf.text() == kf
+    if (not sys.platform.startswith('linux') or
+            not os.environ.get('CI') is not None):
+        assert new_dlg.kfp.text() == kfp
+
+
+def test_kernel_connection_dialog_doesnt_remember_input_ssh_passphrase(
```

Comment:
Remove at least ``input``

Reply:
test_kernel_connection_dialog_doesnt_remember_input_ssh_passphrase is misleading since it remembers more than just the ssh passphrase, but this is the case where it also remembers ssh keyphrase

Reply:
Maybe I'd remove the "kernel_connection" part looking at the names of other test cases in the project.

Reply:
I think you should leave "connection", so its clear and explicit at a glance as to which of Spyder's dozens of dialogs its referring to, but I agree with nixing "kernel". Also, it reads as "Test that the kernel connection dialog doesn't remember the input SSH password", (I *just now* finally got that there was supposed to be a break after input) so you could add "with" after "input" to make that clear.

Reply:
ok done